Source/EventFlow.MsSql/ReliablePublish/MsSqlReliablePublishPersistence.cs

```diff
@@ -0,0 +1,132 @@
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EventFlow.Aggregates;
+using EventFlow.Core;
+using EventFlow.EventStores;
+using EventFlow.PublishRecovery;
+using EventFlow.Subscribers;
+
+namespace EventFlow.MsSql.ReliablePublish
+{
+    public sealed class MsSqlReliablePublishPersistence : IReliablePublishPersistence
+    {
+        private readonly IMsSqlConnection _msSqlConnection;
+
+        public MsSqlReliablePublishPersistence(IMsSqlConnection msSqlConnection)
+        {
+            _msSqlConnection = msSqlConnection;
+        }
+
+        public async Task MarkPublishedAsync(IIdentity aggregateIdentity, IReadOnlyCollection<IDomainEvent> domainEvents)
+        {
+            var item = new PublishLogItem
+            {
+                AggregateId = aggregateIdentity.Value,
+                MinAggregateSequenceNumber = domainEvents.Min(x => x.AggregateSequenceNumber),
+                MaxAggregateSequenceNumber = domainEvents.Max(x => x.AggregateSequenceNumber),
+            };
+
+            await _msSqlConnection.ExecuteAsync(
+                    Label.Named("publishlog-commit"),
+                    CancellationToken.None, // Unable to Cancel
+                    @"INSERT INTO [dbo].[EventFlowPublishLog]
+                    (AggregateId, MinAggregateSequenceNumber, MaxAggregateSequenceNumber)
+                    VALUES
+                    (@AggregateId, @MinAggregateSequenceNumber, @MaxAggregateSequenceNumber)",
+                    item)
+                .ConfigureAwait(false);
+        }
+
+        public async Task<VerificationState> GetUnverifiedItemsAsync(int maxCount, CancellationToken cancellationToken)
+        {
+            var logItems = await _msSqlConnection.QueryAsync<PublishLogItem>(
+                    Label.Named("publishlog-select"),
+                    cancellationToken,
+                    "SELECT TOP(@Top) [Id], [AggregateId],[MinAggregateSequenceNumber], [MaxAggregateSequenceNumber] FROM [dbo].[EventFlowPublishLog] ORDER BY [Id]",
+                    new { Top = maxCount })
+                .ConfigureAwait(false);
+
+            var positions = await _msSqlConnection.QueryAsync<string>(
+                    Label.Named("publishlog-global-position-select"),
+                    cancellationToken,
+                    "SELECT TOP 1 [LastVerifiedPosition] FROM [dbo].[EventFlowPublishVerifyState]")
+                .ConfigureAwait(false);
+
+            return new VerificationState(
+                new GlobalPosition(positions.First()),
+                logItems);
+        }
+
+        public async Task MarkVerifiedAsync(
```

Comment:
Same here — we are leaking implementation detail of having EventFlowPublishVerifyState into interface. 
May be could just call `MarkPublishedAsync` from verificator and have some kind of `Compact` method which is implementation detail and could be MS SQL specific.